run/test/specs/message_checks_groups_android.spec.ts

```diff
@@ -0,0 +1,546 @@
+  const testGroupName = "Message checks for groups";
+  const { device1, device2, device3 } = await openAppThreeDevices(platform);
+  // Create users A, B and C
+  const [userA, userB, userC] = await Promise.all([
+    newUser(device1, "Alice", platform),
+    newUser(device2, "Bob", platform),
+    newUser(device3, "Charlie", platform),
+  ]);
+  // Create contact between User A and User B
+  await createGroup(
+    platform,
+    device1,
+    userA,
+    device2,
+    userB,
+    device3,
+    userC,
+    testGroupName
+  );
+  const testLink = `https://nerdlegame.com/`;
+  // Send a link
+  await device1.inputText("accessibility id", "Message input box", testLink);
+  // Accept dialog for link preview
+  await device1.clickOnElement("Enable");
+  // No preview on first send
+  await device1.clickOnElement("Send message button");
+  await device1.waitForTextElementToBePresent({
+    strategy: "accessibility id",
+    selector: "Message sent status: Sent",
+    maxWait: 20000,
+  });
+  await device2.waitForTextElementToBePresent({
+    strategy: "accessibility id",
+    selector: "Message body",
+    text: testLink,
+  });
+  await device3.waitForTextElementToBePresent({
+    strategy: "accessibility id",
+    selector: "Message body",
+    text: testLink,
+  });
+  await device2.replyToMessage(userA, testLink);
+  await device3.waitForTextElementToBePresent({
+    strategy: "accessibility id",
+    selector: "Message body",
+    text: `${userA.userName} message reply`,
+  });
+  await closeApp(device1, device2, device3);
+}
+
+async function sendGifGroup(platform: SupportedPlatformsType) {
+  const testGroupName = "Message checks for groups";
+  const { device1, device2, device3 } = await openAppThreeDevices(platform);
+  // Create users A, B and C
+  const [userA, userB, userC] = await Promise.all([
+    newUser(device1, "Alice", platform),
+    newUser(device2, "Bob", platform),
+    newUser(device3, "Charlie", platform),
+  ]);
+  // Create contact between User A and User B
+  await createGroup(
+    platform,
+    device1,
+    userA,
+    device2,
+    userB,
+    device3,
+    userC,
+    testGroupName
+  );
+  const replyMessage = `Replying to GIF from ${userA.userName}`;
+  // Click on attachments button
+  await device1.clickOnElement("Attachments button");
+  // Select GIF tab
+  await device1.clickOnElement("GIF button");
+  await device1.clickOnElementAll({
+    strategy: "accessibility id",
+    selector: "Continue",
+  });
+  // Select gif
+  await sleepFor(3000);
+  await device1.clickOnElementXPath(
+    `/hierarchy/android.widget.FrameLayout/android.widget.LinearLayout/android.widget.FrameLayout/android.widget.FrameLayout/android.widget.FrameLayout/android.widget.RelativeLayout/android.widget.ScrollView/androidx.viewpager.widget.ViewPager/android.widget.FrameLayout/androidx.recyclerview.widget.RecyclerView/android.widget.FrameLayout[1]`
+  );
+  // Reply to message
+  await sleepFor(5000);
```

Comment:
why sleepFor

Reply:
Deleted sleepFor and replaced the next function after with waitFor function that have a maxWait loop